### ISSUE TITLE
Dealing with newer laptop boards. 

### ISSUE THREAD

**Issue by popalltheshells** [ufrisk/pcileech#285]
Hi, just curious - has anyone encountered a case where the laptop we use it against only has 1 NVME m.2 slot; and a soldered on WLAN m.2 slot? I was looking for a solution that maybe has an m.2 slot expansion to turn the one slot into two. 

Since PCI Squirrel does need the laptop to be on -> meaning we can't just replace the NVME SSD with the PCI Squirrel. 

How would you resolve this without a PCIe to thunderbolt? Or is that the only way.

**Comment by dmgarry** [comment]
This is something that has crossed my mind before but unfortuanately I havent had any time to test my ideas.

Idea 1:

place m2 ssd into a usb adapter and connect this then via usb/usb-c slot and PCI Squirrel then into internal m2 slot.

Idea 2:

Using an adapter key that goes to a dual m2 card to connect the main ssd and the pci squirrel to and put them in through the same m2 slot on the board.

once again these are just ideas and hope they make sense, they may work or may trigger bitlocker or not work due to 100 other reasons... 😃

**Comment by popalltheshells** [comment]
> Using an adapter key that goes to a dual m2 card to connect the main ssd and the pci squirrel to and put them in through the same m2 slot on the board.

I was looking into this but I can't seem to find an adapter for this; scoured through aliexpress and others. But the m.2 to USB and set the BIOS boot order may work actually, didn't think of that!

**Comment by dmgarry** [comment]
depending on your key this is what i came across before:

https://www.reichelt.com/ie/en/adapter-m-2-key-m-sff-8643-nvme-horizontal-2242-delock-63918-p236856.html?PROVID=2854&gclid=Cj0KCQiAwP6sBhDAARIsAPfK_wb10pMpGxhw0belrsdg1dWHYEuNwfqlLsfjm4dck37I40h0BP_Xj4oaArlQEALw_wcB

![ex1](https://github.com/ufrisk/pcileech/assets/110166362/af7ad1b8-29dc-4033-9e1a-e09fd52f859c)

**Comment by popalltheshells** [comment]
I've gotten myself an NVME m.2 to USB so I can change BIOS boot order to that. From Amazon; will update this thread if it works. Fingers corssed.

**Comment by popalltheshells** [comment]
I tried connecting my PCI squirrel to internal NVME m.2 slot but it's not powering on.. Any ideas? @ufrisk @dmgarry

**Comment by dmgarry** [comment]
you connected external power to adapter cable?

**Comment by popalltheshells** [comment]
I have the squirrel so it should power from PCIe connection - no external power slot from what I can see.

**Comment by dmgarry** [comment]
![adapter](https://github.com/ufrisk/pcileech/assets/110166362/3e64c90d-31c2-4705-8532-4300f920371d)

red circle on the adapter cable shows where external power must be connected

squirrel will only get power directly if connected to pci slot, m2 slot needs external power to adapter cable.

**Comment by popalltheshells** [comment]
OHHHH NO way, okay let me try that! Thank you very much

**Comment by dmgarry** [comment]
I took a power supply something similar from an old pc that has alot of different power connectors on it and keep it in my kit ![powersupply](https://github.com/ufrisk/pcileech/assets/110166362/98263224-fda5-479f-b0f4-9c0986e2e8fb)

**Comment by popalltheshells** [comment]
Turns out my PCIe x4 to M.2 connector has 4 pins (molex connectors?) Like the one your circled; but the cable that came with it only has two pins. 
![image](https://github.com/ufrisk/pcileech/assets/6753178/5431cd57-5cff-4d0f-8b60-b8c75635eb5f)


I was thinking if something like this would work?
![image](https://github.com/ufrisk/pcileech/assets/6753178/714c635a-266a-4ad7-a437-055b2ab2bbde)

**Comment by dmgarry** [comment]
yep that'll connect to the adapter, just need to get power to the other end and it'll power the squirrel

**Comment by popalltheshells** [comment]
Thank you! I'm still wondering why this seller is bundling power cable with a different adapter that won't work (based on your photo) - which was the one I got. 
![image](https://github.com/ufrisk/pcileech/assets/6753178/bcbd744e-f354-4f69-835b-7fd881c96b24)

**Comment by dmgarry** [comment]
You sure the power connector you got with it won't fit? They usually send the right one

**Comment by popalltheshells** [comment]
Yeah the power connector that came with the m.2 (m key) adapter has 2 slots for the pins, but the m.2 adapter itself has 4 pins (I tried connecting it anyways but that didn't work); 

I also happen to have another m.2 adapter (but b key),that only has 2 pins - and it worked with the power 2 pins connector (Squirrel turned on). lol - it is wild

**Comment by popalltheshells** [comment]
Update: I've tried with surface pro 4 - tried plugging the internal NVME to USB and boot from there, and have my PCILeech through the built in NVME slot - both LEDs turned on which is good - but surface pro 4 won't let me boot from USB and kept saying "no OS". 

Have you had success in the past with using PCI screamer on an m.2 NVME slot with the adapter? If so, I'll find another laptop to test it on, may just be the surface pro 4 that I'm using is broken in a way @dmgarry

**Comment by joshuanutt** [comment]
Having the same issues with a Latitude 5420.   External NVME is detected but refuses to boot.
